### PR TITLE
SDIT-1440 Nomis prisoner API duplicate entities

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt [hunk 1]
```diff
@@ -11,15 +11,15 @@ import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
...
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderProgramProfileRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.PrisonIepLevelRepository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
 import java.time.LocalDate
 
 @Service
 @Transactional
 class PayRatesService(
-  private val availablePrisonIepLevelRepository: AvailablePrisonIepLevelRepository,
+  private val availablePrisonIepLevelRepository: PrisonIepLevelRepository,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt:22]: I renamed AvailablePrisonIepLevelRepository to PrisonIepLevelRepository and removed PrisonIncentiveLevel

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesService.kt [hunk 2]
```diff
@@ -297,24 +295,26 @@ class IncentivesService(
...
+            id = createRequest.levelCode,
+            agencyLocation = prison,
             active = createRequest.active,
             default = createRequest.defaultOnAdmission,
             remandTransferLimit = createRequest.remandTransferLimitInPence?.toPounds(),
             remandSpendLimit = createRequest.remandSpendLimitInPence?.toPounds(),
             convictedTransferLimit = createRequest.convictedTransferLimitInPence?.toPounds(),
             convictedSpendLimit = createRequest.convictedSpendLimitInPence?.toPounds(),
             expiryDate = if (createRequest.active) null else LocalDate.now(),
+            iepLevel = iepLevel,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesService.kt:317]: the new PrisonIepLevel has 2 mappings for iepLevel, both code and entity

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/PrisonIepLevel.kt [hunk 3]
```diff
@@ -15,11 +15,13 @@ import org.hibernate.annotations.JoinColumnsOrFormulas
...
 
 @Entity
-@IdClass(AvailablePrisonIepLevel.Companion.PK::class)
+@IdClass(PrisonIepLevel.Companion.PK::class)
 @Table(name = "IEP_LEVELS")
-data class AvailablePrisonIepLevel(
+data class PrisonIepLevel(
 
   @Id
   @Column(name = "IEP_LEVEL")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/PrisonIepLevel.kt:27]: A real shame this attribute is called `id` given it is just part of the real ID. I had the read the HQL below several times before I understood what was going on.

Any chance it could be renamed in a separate pr?

**Review by andymarke — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/PrisonIepLevel.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/PrisonIepLevel.kt:27]: Yep sure
